### DISCUSSION TITLE
Help setting up RSocket Broker within Kubernetes Cluster

### DISCUSSION THREAD

**Discussion by AXTH93** [alibaba/alibaba-rsocket-broker#245] [Q&A]
Hi @linux-china love the work you've done ... I followed many of your projects and using the OpenAI you made, simply amazing.

I have soo many questions to this RSocket Broker, since I really find it difficult to setup. Is there a tutorial out there somewhere which i missed? 

Essentially we have a message-service which can of course create multiple instances... and these services contains chatrooms (sinks with users who are subscribed).... We need to somehow be able to figure out if users are connected to a different message-service instances, then they should still be able to receive messages in-between, so somehow dynamically we want to based on the load, route messages accross one service to another so users who are in the same chatroom but on a different message-service can send and receive messages from and to other users sitting on other message-services with the same chatroom Sink. 

I really feel like this RSocket broker could be the trick since i feel a huge overhead relying on Redis for this ... and using RSocket Requester onan infinite amount of chatrooms seems to be to heavy ... 

Would you be able to help somehow showing me how I could solve this? I know this is a lot to ask, but I'll take my chances and hope for some assistance... just how I implement the broker, and how i can get one message-service to communicate to another through the broker. 

Best Regards 
Alex Holmberg

**Comment by code-uri** [comment]
Hi Alex, Why not just always routes users to the same message service instance?

I understand that if the load is high on certain chat rooms, you wanted to utilise the other instances to distribute the load. May be you can just scale the message service instance with high load to newer instance with more resources.

**Reply by AXTH93** [reply]
Hi thanks for answering, but that is not a scalable good solution if you ask me ... The best solution would be multiple pods being able to inter-communicate whenever users end up in sinks which has the same chatroom as another instance of the same service has created. 

- We want to have horizontally scaled pods based on the load, so having multiple message-services is the most optimal solution, but we don't know how to implement the rsocket broker, to allow these pods communicating with each other.
